### PR TITLE
Fix implicit aliasing and other errors

### PR DESCRIPTION
**Pull request by shyamradhakrishnan**
**What this PR does / why we need it**:
Fix implicit aliasing and other gosec errors

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

**Comment by shyamradhakrishnan** [comment]
e2e test
```
[SynchronizedAfterSuite] PASSED [0.000 seconds]
[SynchronizedAfterSuite]
/home/ubuntu/go/src/github.com/oracle/cluster-api-provider-oci/test/e2e/e2e_suite_test.go:254
------------------------------
[SynchronizedAfterSuite] PASSED [2230.293 seconds]
[SynchronizedAfterSuite]
/home/ubuntu/go/src/github.com/oracle/cluster-api-provider-oci/test/e2e/e2e_suite_test.go:254

  Timeline >>
  STEP: Tearing down the management cluster @ 05/05/23 07:10:56.329
  << Timeline
------------------------------

Ran 6 of 25 Specs in 3896.963 seconds
SUCCESS! -- 6 Passed | 0 Failed | 0 Pending | 19 Skipped
```

**Comment by shyamradhakrishnan** [comment]
unit test
```
?   	github.com/oracle/cluster-api-provider-oci	[no test files]
ok  	github.com/oracle/cluster-api-provider-oci/api/v1beta1	41.611s	coverage: 24.1% of statements
ok  	github.com/oracle/cluster-api-provider-oci/api/v1beta2	0.046s	coverage: 15.8% of statements
?   	github.com/oracle/cluster-api-provider-oci/cloud/config	[no test files]
ok  	github.com/oracle/cluster-api-provider-oci/cloud/ociutil	0.007s	coverage: 19.6% of statements
ok  	github.com/oracle/cluster-api-provider-oci/cloud/scope	235.791s	coverage: 77.1% of statements
?   	github.com/oracle/cluster-api-provider-oci/cloud/scope/mocks	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/base	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/base/mock_base	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/compute	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/compute/mock_compute	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/computemanagement	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/computemanagement/mock_computemanagement	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/containerengine	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/containerengine/mock_containerengine	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/identity	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/identity/mock_identity	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/loadbalancer	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/loadbalancer/mock_lb	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/networkloadbalancer	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/networkloadbalancer/mock_nlb	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/vcn	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/cloud/services/vcn/mock_vcn	[no test files]
ok  	github.com/oracle/cluster-api-provider-oci/cloud/util	0.026s	coverage: 55.8% of statements
ok  	github.com/oracle/cluster-api-provider-oci/controllers	26.119s	coverage: 67.5% of statements
ok  	github.com/oracle/cluster-api-provider-oci/exp/api/v1beta1	27.288s	coverage: 21.8% of statements
ok  	github.com/oracle/cluster-api-provider-oci/exp/api/v1beta2	0.022s	coverage: 10.9% of statements
ok  	github.com/oracle/cluster-api-provider-oci/exp/controllers	1.195s	coverage: 56.3% of statements
?   	github.com/oracle/cluster-api-provider-oci/feature	[no test files]
?   	github.com/oracle/cluster-api-provider-oci/version	[no test files]
```

File: cloud/scope/machine.go [hunk 1]
```diff
@@ -41,12 +37,15 @@ import (
 	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/klog/v2/klogr"
 	"k8s.io/utils/pointer"
+	"math/big"
+	"net/url"
```
Comment (joekr) [inline cloud/scope/machine.go:41]: can you move all the Go internal packages up?

File: cloud/scope/machine.go [hunk 2]
```diff
@@ -195,10 +194,13 @@ func (m *MachineScope) GetOrCreateMachine(ctx context.Context) (*core.Instance,
...
-		rand.Seed(time.Now().UnixNano())
-		// rand.Intn(3) will produce a random number from 0(inclusive) to 3(exclusive)
-		// ee add one to get a number from 1 to 3
-		failureDomain = common.String(strconv.Itoa(rand.Intn(3) + 1))
+		randomFaultDomain, err := rand.Int(rand.Reader, big.NewInt(3))
+		if err != nil {
+			m.Logger.Error(err, "Failed to generate random fault domain")
+			return nil, err
+		}
+		// the random number generated is between zero and two, whereas we need a number between one and three
```
Comment (joekr) [inline cloud/scope/machine.go:202]: Awesome comment!

File: api/v1beta1/ocicluster_conversion.go [hunk 3]
```diff
@@ -86,9 +86,10 @@ func convertv1beta2NSGListTov1beta1NSGList(in []*v1beta2.NSG) ([]*NSG, error) {
 
 func Convertv1beta1AdMapTov1beta2AdMap(in map[string]OCIAvailabilityDomain) (map[string]v1beta2.OCIAvailabilityDomain, error) {
 	out := make(map[string]v1beta2.OCIAvailabilityDomain)
-	for k, v := range in {
+	for k, _ := range in {
 		outV := &v1beta2.OCIAvailabilityDomain{}
-		err := Convert_v1beta1_OCIAvailabilityDomain_To_v1beta2_OCIAvailabilityDomain(&v, outV, nil)
+		inDomain := in[k]
```
Comment (joekr) [inline api/v1beta1/ocicluster_conversion.go:91]: more of a question than a request for change: why not use `&in[k]` in the below call like what is done in cloud/scope/machine_pool.go

**Review by joekr — CHANGES_REQUESTED** [review]
Super small NIT

**Pull request closed without merging**